web/TrackPlayer/PlaylistPlayer.ts

```diff
@@ -10,8 +11,15 @@ export class PlaylistPlayer extends Player {
   protected _currentIndex?: number;
   protected repeatMode: RepeatMode = RepeatMode.Off;
 
+  protected async onStateUpdate(state: Exclude<State, State.Error>) {
+    super.onStateUpdate(state);
```

Comment:
This will still emit a state: ended, right? even though we might not be at the end of the queue.

Reply:
I guess we want to delay this call, and only send it out in `onPlaylistEnded`

Reply:
@dcvz I believe this is right, isn't it? Here's the sequence:

1. The track will end.
2. An `Event.PlaybackState` with a value of `State.Ended` will be emitted signaling the track has ended.
3. We'll run the `onTrackEdned/onPlaylistEnded` methods which will emit an `Event.PlaybackQueueEnded` event signaling the queue has ended.

So we get an event saying that the track has ended, and another event that says the queue has ended. Are we only supposed to emit one or the other?

Reply:
This is what iOS does: https://github.com/doublesymmetry/react-native-track-player/blob/ef9fb0061ce5fc6c2892bbc480482a832ac6af92/ios/RNTrackPlayer/RNTrackPlayer.swift#L806

This is what Android does: https://github.com/doublesymmetry/react-native-track-player/blob/ef9fb0061ce5fc6c2892bbc480482a832ac6af92/android/src/main/java/com/doublesymmetry/trackplayer/service/MusicService.kt#L606


If I'm following the logic correctly here, it seems like this should have the same outcome! 👍